tests/functional/externalpages/analytics/test_google_analytics.py

```diff
@@ -0,0 +1,49 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import pytest
+
+from pages.externalpages.analytics.index import AnalyticsTestPage
+
+
+def assert_ga_loaded(selenium):
+    # if the datalayer list doesnt contain the gtm.load event then GA has not been loaded or it has been blocked
+    data_layer = selenium.execute_script("return window.dataLayer")
+    assert any(
+        "event" in layer and layer["event"] == "gtm.load" for layer in data_layer
+    ), "These tests require Google Analytics to be configured. \
+       Please ensure you have set a GTM_CONTAINER_ID environment variable in your .env and you are not blocking GA in your browser."
+
+
+@pytest.mark.nondestructive
+def test_download_button(base_url, selenium):
+    page = AnalyticsTestPage(selenium, base_url).open()
+    assert_ga_loaded(selenium)
+    assert page.download_button_is_displayed
+    page.click_download_button()
+    data_layer = selenium.execute_script("return window.dataLayer")
+    link_type = page.download_button.get_attribute("data-link-type")
+    assert any("gtm.element" in layer and layer["gtm.element"].get_attribute("data-link-type") == link_type for layer in data_layer)
+
+
+@pytest.mark.nondestructive
+def test_link_button(base_url, selenium):
+    page = AnalyticsTestPage(selenium, base_url).open()
+    assert_ga_loaded(selenium)
+    assert page.link_button_is_displayed
+    page.click_link_button()
+    data_layer = selenium.execute_script("return window.dataLayer")
+    cta_type = page.link_button.get_attribute("data-cta-type")
+    assert any("gtm.element" in layer and layer["gtm.element"].get_attribute("data-cta-type") == cta_type for layer in data_layer)
+
+
+@pytest.mark.nondestructive
+def test_account_button(base_url, selenium):
+    page = AnalyticsTestPage(selenium, base_url).open()
+    assert_ga_loaded(selenium)
+    assert page.account_button_is_displayed
+    page.click_account_button()
+    data_layer = selenium.execute_script("return window.dataLayer")
+    cta_type = page.account_button.get_attribute("data-cta-type")
+    assert any("gtm.element" in layer and layer["gtm.element"].get_attribute("data-cta-type") == cta_type for layer in data_layer)
```

Comment:
Just wanted to point out the difference between `any` and `all`. The `any` call will pass if any single item in the list passes the test. If you want to require all items in the list to pass the test then `all` can be used.